extensions/ql-vscode/src/codeql-cli/cli.ts

```diff
@@ -362,6 +362,8 @@ export class CodeQLCliServer implements Disposable {
     silent?: boolean,
   ): Promise<string> {
     const stderrBuffers: Buffer[] = [];
+    // The array of fragments of stderr of this line. To be used for logging.
+    let currentLineStderrBuffers: Buffer[] = [];
```

Comment:
Would it be simpler to store a single `currentLineBuffer: Buffer` instead of an array of buffers?

Currently, whenever we want to print the current line we have to concatenate all the buffers together, but this could be avoided if we stored the concatenated buffer. And I don't think it would make for extra work when adding a new data fragment. The current process is:
- add new fragment to `currentLineStderrBuffers` array
- Concatenate `currentLineStderrBuffers` array into `currentLineBuffer` and check if it contains an `EOL`
- After handling any newlines, set `currentLineStderrBuffers = [currentLineBuffer]`. Maintaining the original individual fragments isn't important and after handling any newlines that data is lost.

I think we could simplify so whenever any new fragment comes in we do `currentLineBuffer = Buffer.concat([currentLineBuffer, newData])` and then proceed as before, updating `currentLineBuffer` whenever we find a `EOL` and print a line.